### DISCUSSION TITLE
chimney-protobuf: To have a (default?) configuration setting allowing to skip manual assign for UnknownFieldSet

### DISCUSSION THREAD

**Discussion by codesurf42** [scalalandio/chimney#506] [Ideas]
Chimney is getting better with full ADT conversions 🚀 . One small issue when using it with Protobuf is https://chimney.readthedocs.io/en/stable/cookbook/#unknownfieldset While switching this setting off simplify chimney conversions and chances of getting false-positive conversions with default values, unknown fields seem to be the default in proto and to stay there for a long time (https://protobuf.dev/programming-guides/proto3/#unknowns, off and on rollback in proto3)  and in many projects changing these default proto settings may not be the best way to go.

While someone can just add a manual conversion with `.withFieldConst(_.unknownFields, UnknownFieldSet())` (and similar for ADTs), maybe an easier, "out of the box" way would be to have such default conversion in place when also using `chimney-protobuf` module?

**Comment by MateuszKubuszok** [comment]
In general - yes it would be convenient.

Problem is that:

 * protobufs are an additional module
 * core Chimney is as dependency-free as possible - currently, it depends on the standard library (unavoidable), `scala-reflect` on Scala 2 (necessary for macros) and `scala-collection-compat` (to handle 2.12 vs 2.13+ differences in collections)
 * so macros in core Chimney know nothing about the types defined in Cats or ScalaPB
 * the best way would be to somehow handle that with a config that could be injected into macro, where `UnknownFieldSet` would be just an example

We have https://github.com/scalalandio/chimney/issues/408 created for that reason, but so far whenever there was some time for development, there were more important things to fix (mostly because some things needs to be cleaned up before committing to 1.0.0 and semver, since they break bincompat, and some can be fixed later without breaking changes).

**Reply by MateuszKubuszok** [reply]
I forgot about this but, #408 got implemented:

 * protobuf module provides default `UnknownFieldSet` - https://chimney.readthedocs.io/en/stable/cookbook/#custom-default-values
 * so when one compiles protobufs with default values **disabled** but **enable them in Chimney** this integration would provide the value for the field.

Beyond that, I doubt that we would be able to provide any more support ever.
